### ISSUE TITLE
bug: ion-loading has no aria-label attribute

### ISSUE THREAD

**Issue by ptmkenny** [ionic-team/ionic-framework#24486]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

`ion-loading` does not have the `aria-label` attribute set, and although the `<IonLoading>` component accepts an `aria-label` prop in React, the `aria-label` does not get set.

axe identifies the issue as: `ARIA dialog and alertdialog nodes must have an accessible name`

### Expected Behavior

`ion-loading` should set the aria-label or `title`, or provide the developer with a way to do so.

### Steps to Reproduce

Use `ion-loading` in an app.  Check with [@axe-core/react](https://github.com/dequelabs/axe-core-npm/blob/develop/packages/react/README.md) or a similar tool.

### Ionic Info

Ionic:

   Ionic CLI       : 6.18.1
   Ionic Framework : @ionic/react 6.0.1

Capacitor:

   Capacitor CLI      : 3.3.3
   @capacitor/android : 3.3.3
   @capacitor/core    : 3.3.3
   @capacitor/ios     : 3.3.3

Utility:

   cordova-res : 0.15.4
   native-run  : 1.5.0

System:

   NodeJS : v16.13.1 (/usr/bin/node)
   npm    : 8.1.2
   OS     : Linux 5.10

### Additional Information

Similar to this issue: https://github.com/ionic-team/ionic-framework/issues/23108

**Comment by amandaesmith3** [comment]
Thanks for the issue. We should do something similar to how `ion-menu` handled this: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/menu/menu.tsx#L703

**Comment by sean-perkins** [comment]
I think we need to initially uncover why `ion-loading` is one of the only overlay implementations that has the role set on the inner container (`role="dialog"`), but doesn't have `aria-modal` set on the host. 

The introduction of `htmlAttributes` as an assignable property _should have_ supported the ability to pass the necessary aria attributes to the host, but the nested declaration causes the AXE issues.

If we can remove the aria-role from the inner container and append it to the host, without regressing accessibility testing, that would be my recommendation.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/26581, and a fix will be available in an upcoming release of Ionic Framework.
